packages/@expo/cli/src/start/server/metro/inspector-proxy/__tests__/proxy.test.ts

```diff
@@ -0,0 +1,33 @@
+import { InspectorProxy as MetroProxy } from 'metro-inspector-proxy';
+
+import { ExpoInspectorProxy as ExpoProxy } from '../proxy';
+
+describe(ExpoProxy, () => {
+  it('shares devices with metro proxy', () => {
+    const { expoProxy, metroProxy } = createTestProxy();
+    expect(metroProxy._devices).toBe(expoProxy.devices);
+  });
+
+  it('forwards `processRequest` to metro proxy', () => {
+    const { expoProxy, metroProxy } = createTestProxy();
+    const spy = jest.spyOn(metroProxy, 'processRequest');
+    const ctx = {
+      req: jest.fn() as any,
+      res: jest.fn() as any,
+      next: jest.fn(),
+    };
+
+    expoProxy.processRequest(ctx.req, ctx.res, ctx.next);
+    expect(spy).toBeCalledWith(ctx.req, ctx.res, ctx.next);
+  });
+
+  // TODO(cedric): add tests for `createWebSocketListeners`, using a mocked websocket
```

Comment:
@EvanBacon We probably need something like https://www.npmjs.com/package/jest-websocket-mock for these tests to be effective.

Alternatively, I could split up the device initialization and test if devices are created successfully from URL. Unfortunately, that won't test the `.on('connection')` and `.on('close')` events.

Any thoughts on this one?

Reply:
the helper seems fine, I've never used it before.